src/main/java/com/thealgorithms/geometry/GrahamScan.java

```diff
@@ -20,17 +19,12 @@ public class GrahamScan {
     private final Stack<Point> hull = new Stack<>();
 
     public GrahamScan(Point[] points) {
```

Comment:
It seems that this implementation does not validate the input data, that the minimum number of points must be 3 to construct a convex hull. And also if all the points are the same, then it also would not be able to process such input. 

Checks can be look something like this:

```
if (points == null || points.length < 3) {
    throw new IllegalArgumentException("At least 3 points are required to compute the convex hull");
}

// Check if all points are the same
boolean allSame = true;
for (int i = 1; i < points.length; i++) {
    if (!points[i].equals(points[0])) {
        allSame = false;
        break;
    }
}

if (allSame) {
    hull.push(points[0]);
    return;
}
```

It is also good to have appropriate tests for such cases.

Reply:
added 👍

Reply:
@alxkm what is wrong in computing the convex hull of two points, one point or even zero points? In such case (if the points are in the _general position_) the output should be the same as the input.

If some of the input points are _the same_, they are simply _collinear_, and the algorithms should handle that.

Reply:
@vil02, I thought that the convex hull (CH) of two points is problematic because a CH requires at least three non-collinear points to form a valid polygon, and CH is defined as the smallest convex polygon that can enclose a set of points. With only two points, you can't form a polygon, you can only form a line segment.

But there are two points of view. From mathematical perspective, and a standard expectations from CH is. 

From mathematical perspective all these cases (0, 1, 2) are trivial, and return just themselves. But I left comment from result perspective that we expect that algorithm just forming a polygon.

Thanks for clarification. I understand that we support mathematical point of view here (from logical perspective it is only right).

Reply:
@bayramtturgutt sorry for misunderstanding, but this suggestion for checking that points are the same, and input expectations for 3 points are wrong.  

Could you please revert these changes.

Reply:
No worries, I'll revert those changes :)